spec/formatting.md

```diff
@@ -14,9 +14,11 @@ To resolve the value of a Variable,
 its Name is used to identify either a local variable,
 or a variable defined elsewhere.
 If a local variable and an externally defined one use the same name,
-the local variable takes precedence.
+the local variable takes precedence after its variable declaration.
+A variable declaration may refer in its expression
+to a previous or external variable with the same name as the declaration's target.
 
-It is an error for a local variable definition to
+It is an error for a variable declaration to
 refer to a local variable that's defined after it in the message.
```

Comment:
```suggestion
It is an error for a `declaration` to refer to an undefined variable.
```

This statement is trickier than it looks. A `declaration` can refer to an external variable and there is no list of these. We should probably say which type of error it is?

Reply:
This is covered in the Error Handling section, which specifies that this would be an Unresolved Variable error.

Reply:
Yes, but this requires someone to go read a separate location in the spec to find out what type of error it is. We don't need to define error handling here, but it's friendlier to readers to say:

> A `declaration` that refers to an undefined variable results in an `Unresolved Variable error`. 
> Local variables **_must_** be defined before any `declaration` or `expression` that refers to them.